scripts/check_icons.sh

```diff
@@ -2,44 +2,83 @@
 
 set -e
 
-check_icon() (
+check_geometry() (
     error=0
     file="$1"
-    geometry="$2"
-    content=$(identify -verbose "$file")
-
-    if [[ "$geometry" != "no-check" ]]; then
-        if ! echo "$content" | grep -q "Geometry: $geometry"; then
-            echo "ERROR => Icon should have a $geometry geometry"
-            error=1
-        fi
-    fi
+    device="$2"
+
+    case "$device" in
+        "nanos")
+            geometry="16x16"
+            ;;
+        "nanox")
+            geometry="14x14"
+            ;;
+        "nanos2")
+            geometry="14x14"
+            ;;
+        "stax")
+            geometry="32x32"
+            ;;
+        *)
+            echo "ERROR => Device '$device' not recognized"
+            return 1
+            ;;
+    esac
 
-    if echo "$content" | grep -q "Alpha"; then
-        echo "ERROR => Icon should have no alpha channel"
+    if ! identify -verbose "$file" | grep -q "Geometry: $geometry"; then
+        echo "ERROR => Icon '$file' for '$device' should have a '$geometry' geometry"
         error=1
+    else
+        echo "SUCCESS => Icon '$file' for '$device' has a '$geometry' geometry"
     fi
 
-    if ! echo "$content" | grep -q "Colors: 2"; then
-        echo "ERROR => Icon should have only 2 colors"
-        error=1
+    return "$error"
+)
+
+check_glyph() (
+    error=0
+    file="$1"
```

Comment:
Can you add a log here with something like:
`"Checking glyph file $file"`

Reply:
Done